src/main.js

```diff
@@ -791,14 +791,16 @@ Apify.main(async () => {
                         ),
                     ]);
 
+                    totalResults = Math.max(results.length, totalResults);
+
                     if (!results?.length) {
                         session.retire();
                         if (totalCount > 0) {
                             await Apify.setValue(`SEARCHSTATE-${Math.random()}`, queryStates);
                             throw new Error(`No map results but result count is ${totalCount}`);
                         } else {
                             log.debug('Really zero results');
-                            return;
+                            throw new Error(`Zero results found. Retry request.`);
```

Comment:
this is actually expected, retrying usually don't yield results the next time. it fails to load before reaching here

Reply:
@pocesar The problem might originate somewhere else (page not loaded fully?) but when I tested the actor with a simple input, it failed in 30-40 % cases and stopped scraping after extracting about 40 items (or better said, it didn't fail, it just finished much earlier than it should without throwing any error). So I traced where it stops running and returns from the `main` function and got into this condition. Then I tested many runs with throwing an error in that place and retrying the request helped in each of these runs. Actor kept scraping more items normally. From my observation, the problem occurred at the beginning of scraping so it might really be some captcha blocking the session before it settles with the right proxy. Any idea how to deal with this?

My test input: 

```json
{
    "search": "Los Angeles",
    "type": "all",
    "maxLevel": 5,
    "maxItems": 200,
    "simple": true,
    "proxyConfiguration": {
      "useApifyProxy": true
    },
    "maxRetries": 20,
    "extendOutputFunction": "async ({ data, item, customData, Apify }) => {\n    return item;\n}",
    "extendScraperFunction": "async ({ label, page, request, customData, Apify }) => {\n    if (label === 'SETUP') {\n        // before crawler.run()\n    } else if (label === 'GOTO') {\n        // inside handleGotoFunction\n    } else if (label === 'HANDLE') {\n        // inside handlePageFunction\n    } else if (label === 'FINISH') {\n        // after crawler.run()\n    }\n}",
    "handlePageTimeoutSecs": 3600,
    "stealth": false,
    "debugLog": false,
    "customData": {}
}
```